### PR TITLE
Refactor ServiceLoader related classes

### PR DESCRIPTION
**Pull request by brfrn169**
This PR refactors the change in #661. I will add inline comments for the details. Please take a look!

### CODE REVIEW

File: core/src/main/java/com/scalar/db/api/DistributedStorageProvider.java [hunk 1]
```diff
@@ -3,15 +3,15 @@
...
- * instances. Each storage adapter should implement this class to instantiate its implementations of
- * {@link DistributedStorage} and {@link DistributedStorageAdmin}. The implementations are assumed
- * to be loaded by {@link java.util.ServiceLoader}.
+ * A class that creates {@link DistributedStorage} and {@link DistributedStorageAdmin} instances.
+ * Each storage adapter should implement this class to instantiate its implementations of {@link
+ * DistributedStorage} and {@link DistributedStorageAdmin}. The implementations are assumed to be
+ * loaded by {@link java.util.ServiceLoader}.
  */
-public interface DistributedStorageFactory {
+public interface DistributedStorageProvider {
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/api/DistributedStorageProvider.java:11]: Renamed `DistributedStorageFactory` to `DistributedStorageProvider`.

File: core/src/main/java/com/scalar/db/api/DistributedTransactionProvider.java [hunk 2]
```diff
@@ -3,17 +3,17 @@
...
- * TwoPhaseCommitTransactionManager}. The implementations are assumed to be loaded by {@link
- * java.util.ServiceLoader}.
+ * A class that creates {@link DistributedTransactionManager}, {@link DistributedTransactionAdmin},
+ * and {@link TwoPhaseCommitTransactionManager} instances. Each transaction manager should implement
+ * this class to instantiate its implementations of {@link DistributedTransactionManager}, {@link
+ * DistributedTransactionAdmin}, and {@link TwoPhaseCommitTransactionManager}. The implementations
+ * are assumed to be loaded by {@link java.util.ServiceLoader}.
  */
-public interface DistributedTransactionFactory {
+public interface DistributedTransactionProvider {
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/api/DistributedTransactionProvider.java:12]: Also renamed `DistributedTransactionFactory` to `DistributedTransactionProvider`

File: core/src/main/java/com/scalar/db/service/ProviderManager.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+import com.scalar.db.config.DatabaseConfig;
+import java.util.Locale;
+import java.util.Map;
+import java.util.ServiceLoader;
+
+/**
+ * This class loads {@link DistributedStorageProvider} and {@link DistributedTransactionProvider}
+ * instances with {@link ServiceLoader} and cache them.
+ */
+final class ProviderManager {
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/service/ProviderManager.java:20]: Also renamed `FactoryManager` to `ProviderManager`.

File: core/src/main/java/com/scalar/db/api/DistributedStorageProvider.java [hunk 4]
```diff
@@ -23,13 +23,13 @@ public interface DistributedStorageFactory {
    * @param config a database config
    * @return an instance of {@link DistributedStorage} for the adapter
    */
-  DistributedStorage getDistributedStorage(DatabaseConfig config);
+  DistributedStorage createDistributedStorage(DatabaseConfig config);
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/api/DistributedStorageProvider.java:26]: Renamed `DistributedStorageFactory` to `DistributedStorageProvider`.

File: core/src/main/java/com/scalar/db/api/DistributedTransactionProvider.java (hunk 2 above)
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/api/DistributedTransactionProvider.java:12]: Also renamed `DistributedTransactionFactory` to `DistributedTransactionProvider`.

File: core/src/main/java/com/scalar/db/service/ProviderManager.java (hunk 3 above)
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/service/ProviderManager.java:20]: Also renamed `FactoryManager` to `ProviderManager`.

File: core/src/main/java/com/scalar/db/api/DistributedStorageProvider.java (hunk 1 above)
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/api/DistributedStorageProvider.java:11]: Renamed `DistributedStorageFactory` to `DistributedStorageProvider`.

**Review by Torch3333 — APPROVED** [review]
LGTM, thank you!

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**